cmd/draft/connect.go

```diff
@@ -96,41 +105,92 @@ func (cn *connectCmd) run(runningEnvironment string) (err error) {
 
 	var connectionMessage = "Your connection is still active.\n"
 
+	exportEnv := make(map[string]string)
+
 	// output all local ports first - easier to spot
 	for _, cc := range connection.ContainerConnections {
 		for _, t := range cc.Tunnels {
-			err = t.ForwardPort()
-			if err != nil {
+			if err = t.ForwardPort(); err != nil {
 				return err
 			}
-			m := fmt.Sprintf("Connect to %v:%v on localhost:%#v\n", cc.ContainerName, t.Remote, t.Local)
-
-			connectionMessage += m
-			fmt.Fprintf(cn.out, m)
-		}
-	}
-
-	for _, cc := range connection.ContainerConnections {
-		readCloser, err := connection.RequestLogStream(deployedApp.Namespace, cc.ContainerName, cn.logLines)
-		if err != nil {
-			return err
+			if export {
+				var (
+					application = sanitize(deployedApp.Name)
+					container   = sanitize(cc.ContainerName)
+					prefix      = fmt.Sprintf("%s_%s", application, container)
+				)
+				exportEnv[prefix+"_SERVICE_HOST"] = fmt.Sprintf("localhost")
+				exportEnv[prefix+"_SERVICE_PORT"] = fmt.Sprintf("%#v", t.Local)
+			} else {
+				m := fmt.Sprintf("Connect to %v:%v on localhost:%#v\n", cc.ContainerName, t.Remote, t.Local)
+				connectionMessage += m
+				fmt.Fprintf(cn.out, m)
+			}
 		}
-
-		defer readCloser.Close()
-		go writeContainerLogs(cn.out, readCloser, cc.ContainerName)
 	}
 
-	stop := make(chan os.Signal, 2)
+	stop := make(chan os.Signal, 1)
+	done := make(chan struct{})
 	signal.Notify(stop, os.Interrupt, syscall.SIGTERM)
 	go func() {
 		<-stop
-		os.Exit(0)
+		close(done)
 	}()
 
-	for {
-		time.Sleep(5 * time.Minute)
-		fmt.Fprintf(cn.out, connectionMessage)
+	if export {
+		exportConnectEnv(exportEnv)
+		os.Stdout.Close()
+		<-done
+		return nil
+	} else {
+		for _, cc := range connection.ContainerConnections {
+			readCloser, err := connection.RequestLogStream(deployedApp.Namespace, cc.ContainerName, cn.logLines)
+			if err != nil {
+				return err
+			}
+			defer readCloser.Close()
+			go writeContainerLogs(cn.out, readCloser, cc.ContainerName)
+		}
+		ticker := time.NewTicker(5 * time.Minute)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				fmt.Fprintf(cn.out, connectionMessage)
+			case <-done:
+				return nil
+			}
+		}
+	}
+}
+
+func (cn *connectCmd) detach() error {
+	args := []string{"connect", "--export"}
+	if len(overridePorts) > 0 {
+		args = append(args, "-p")
+		args = append(args, overridePorts...)
```

Comment:
I think the proper syntax is

```
draft connect -p 8080:8080 -p 8080:8081
```

So you need to iterate through each overridePort and append a `-p` flag rather than just once.

Reply:
so this could be simplified to

```
for i := range overridePorts {
	args = append(args, "-p")
	args = append(args, overridePorts[i])
}
```